gnomad_constraint/plots/plotting_functions.R

```diff
@@ -0,0 +1,377 @@
+    annotate("text", x = .70, y = .20, label = auc2, color = color2)
+
+  return(p)
+}
+
+expected_projections <- function(
+    df,
+    label = "pLoF",
+    sample_size_df = dataset_sample_sizes,
+    xlimits = c(100, 1e8)) {
+  # Plot displaying the percent of genes that would be expected to have a
+  # certain number or variants based on sample size
+  # df: Dataframe with columns 'n_variants', 'n_required', and 'rank'
+  # label: Text that will be included at the top of the plot
+  # sample_size_df: Dataframe with columns 'intercepts', 'linetype', and
+  # 'label' for vertical lines on the plot that show the sample size of some
+  # key datasets.
+  # xlimits: Define the limits of the x-axis
+  # Returns: ggplot object
+  df <- df %>%
+    mutate(
+      n_variants = forcats::fct_reorder(as.factor(.data$n_variants), .data$n_variants)
+    )
+
+  p <- ggplot(df, aes(y = .data$rank, x = .data$n_required, color = .data$n_variants)) +
+    geom_line(linewidth = 2) +
+    theme_classic() +
+    scale_x_log10(labels = comma, limits = xlimits) +
+    xlab("Sample size required") +
+    ylab("Percent of human genes") +
+    geom_vline(
+      data = sample_size_df,
+      aes(xintercept = .data$intercepts, linetype = .data$linetype),
+      key_glyph = "path"
+    )
+
+  # Add manual linetype scale for legend
+  p <- p +
+    scale_linetype_manual(
+      name = "Database size",
+      values = sample_size_df$linetype,
+      labels = sample_size_df$labels
+    ) +
+    scale_color_discrete(name = ">= N variants\nexpected", h = c(40, 120)) +
+    annotate("text", x = xlimits[1], y = 1, hjust = 0, vjust = 1, label = label)
+
+  return(p)
+}
+
+plot_projected_sample_size <- function(df) {
+  # Plot of the percent of genes that have a variable expected number of
+  # variants across sample sizes
+  # df: Dataframe consisting of downsampling data per specific genetic ancestry groups
+  # (has to include 'global')
+  # Returns: ggplot object of the percent of genes that would be expected to have a
+  # certain number or variants based on sample size
+
+  # Filter to rows where a respective gene has at least 1 lof, mis, and syn variant
+  # within all its respective rows
+  # Convert expected counts and n downsamplings to log scale
+  # Fit linear models where expected counts are a function of downsampling size for
+  # each gene
+  df <- df %>%
+    group_by(.data$gene) %>%
+    filter(min(.data$exp_lof) > 0 & min(.data$exp_mis) > 0 & min(.data$exp_syn) > 0) %>%
+    mutate(
+      log_exp_lof = log10(.data$exp_lof),
+      log_exp_mis = log10(.data$exp_mis),
+      log_exp_syn = log10(.data$exp_syn),
+      log_n = log10(.data$downsampling)
+    ) %>%
+    summarize(
+      lof_fit = list(lm(.data$log_exp_lof ~ .data$log_n)),
+      mis_fit = list(lm(.data$log_exp_mis ~ .data$log_n)),
+      syn_fit = list(lm(.data$log_exp_syn ~ .data$log_n))
+    )
+
+  # Extract slope and intercept for lof variants
```

Comment:
oops, I meant to add these back to their original spots after figuring out the formatting. I will do that in the next round